xbmc/music/tags/TagLoaderTagLib.cpp

```diff
@@ -462,6 +464,44 @@ bool CTagLoaderTagLib::ParseTag(ID3v2::Tag *id3v2, EmbeddedArt *art, MUSIC_INFO:
           tag.SetUserrating(POPMtoXBMC(popFrame->rating()));
         }
       }
+    else if (it->first == "CHAP")
+    {
+      //We use a chapter number index here to ensure chapters are added in the right order to the map
+      int chapNumber = 0;
+      unsigned int prevTime = 0;
+      for (ID3v2::FrameList::ConstIterator ch = it->second.begin(); ch != it->second.end(); ++ch)
+      {
+        auto* chapFrame = dynamic_cast<ID3v2::ChapterFrame*>(*ch);
+        if (!chapFrame)
+          continue;
+        std::string embeddedTitle{};
+        std::string chapData = chapFrame->toString().toCString(true);
+        std::vector<std::string> chapInfo = StringUtils::Split(chapData, ",");
+        if (chapFrame->embeddedFrameList().size() > 0)
+          embeddedTitle = chapFrame->embeddedFrameList("TIT2")[0]->toString().toCString(true);
+        unsigned int startTime = chapFrame->startTime(); // times are in millisecs
+        unsigned int endTime = chapFrame->endTime();
+        std::vector<std::string> titles{};
+        if (embeddedTitle.empty())
+          titles = StringUtils::Split(chapInfo[0], ":");
+        // Make sure the last chapter really goes to the end of the file
+        if (prevTime == endTime)
+          endTime = m_totalLen;
+        if (!embeddedTitle.empty()) // prefer an embedded title frame if available
+          chapters[chapNumber].emplace_back(embeddedTitle);
+        else if (!titles[0].empty())
```

Comment:
`else if (!titles.front().empty())` - not 0100% sure but I believe front() is faster than index for vectors

Reply:
Changed, although I'm not sure faster really makes much difference in these days given the amount of processing power in even a tiny prototyping board.

Reply:
Think big. If everybody only looks at their small code pieces, and does not care about optimizations, this in total could make a big difference for the app in a whole.